### PR TITLE
Integrate prettier

### PR DESCRIPTION
**Pull request by vigneshshanmugam**
+ prettier integration with eslint

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,11 +4,11 @@
         "es6": true,
         "node": true
     },
+    "plugins": [
+        "prettier"
+    ],
     "rules": {
-        "brace-style": [2, "1tbs"],
```
Comment (vigneshshanmugam) [inline .eslintrc]: these options are removed from here to match with prettier defaults.

File: .eslintrc [hunk 2]
```diff
@@ -17,15 +17,14 @@
...
         "no-lonely-if": 2,
-        "object-curly-spacing": [2, "always"],
         "quotes": [2, "single"],
-        "semi": [2, "always"],
-        "keyword-spacing": [2],
-        "space-before-blocks": [2, "always"],
-        "strict": [2, "global"]
+        "strict": [2, "global"],
+        "prettier/prettier": [2, {
+            "singleQuote": true,
```
Comment (vigneshshanmugam) [inline .eslintrc]: to match with quotes option in eslint

File: lib/parse-template.js [hunk 3]
```diff
@@ -8,8 +8,14 @@ const Transform = require('./transform');
  * @param {Array} insertBeforePipeTags - Pipe definition will be inserted before these tags
  * @returns {Promise} Promise that resolves to serialized array consisits of buffer and fragment objects
  */
-module.exports = (handledTags, insertBeforePipeTags) =>
-    (baseTemplate, childTemplate, fullRendering = true) => {
-        return Promise.resolve(new Transform(handledTags, insertBeforePipeTags))
-            .then((transform => transform.applyTransforms(baseTemplate, childTemplate, fullRendering)));
-    };
+module.exports = (handledTags, insertBeforePipeTags) => (
```
Comment (vigneshshanmugam) [inline lib/parse-template.js]: god this module looks so much nicer.

File: tests/tailor.js [hunk 4]
```diff
@@ -609,390 +722,431 @@ describe('Tailor', () => {
...
-                    '<script data-pipe>p.end(0, "http://script-link", {"id":0,"range":[0,0]})</script>' +
-                    '</body>' +
-                    '</html>'
-                );
-            }).then(done, done);
+            mockTemplate.returns(
+                '<fragment src="https://fragment/1"></fragment>'
+            );
+
+            getResponse('http://localhost:8081/test')
```
Comment (vigneshshanmugam) [inline tests/tailor.js]: don't like how formatting is done for these functions

**Comment by addityasingh** [comment]
👍

**Comment by vigneshshanmugam** [comment]
👍

**Comment by addityasingh** [comment]
👍

**Pull request merged**